main.go

```diff
@@ -148,13 +146,28 @@ func main() {
 		app.autostartEnabled = true
 	}
 
-	hnsErrCh := make(chan error)
 	serverErrCh := make(chan error)
 	onBoardingFilename := path.Join(app.config.Path, "init")
 	onBoarded := onBoardingSeen(onBoardingFilename)
 
+	hnsdPath, err := getProcPath()
+	if err != nil {
+		log.Fatal(err)
+	}
+	hnsdPath = path.Join(hnsdPath, "/hnsd")
+
+	hnsdCheckpointPath := ""
+	if hnsdPath == "" {
+		log.Fatal("path to hnsd is not provided")
+	}
+	if hnsdCheckpointPath == "" {
+		home, _ := os.UserHomeDir() //above already fails if it doesn't exist
+		hnsdCheckpointPath = path.Join(home, ".hnsd")
+	}
```

Comment:
`hnsdCheckpointPath` has this useless check I copied over a long time ago. The string is always empty, can remove the check.

Reply:
removed